### ISSUE TITLE
Changing notebook path loads new file but shift-enter doesn't work

### ISSUE THREAD

**Issue by box-turtle** [datalayer/jupyter-ui#219]
## Description

I have a <Notebook> component with the path set to an existing file, eg. ping.ipynb. 

In my react component have a useState variable called path initially set to "ping.ipynb". When that file is loaded I can hit shift-enter to execute a cell.

If I change that using setPath("another.ipynb"), the notebook loads the new file, but if I press shift-enter in a cell nothing happens. I can execute the cell using the redux action `dispatch(notebookActions.run.started(NOTEBOOK_UID));`, it is only the shift-enter that does not work.

## Reproduce

1. load a page with a notebook control with the path set to an existing file
2. run a cell using shift-enter - works as expected
3. change the value of the path prop
4. click on a code cell and hit shift-enter
5. nothing happens

Not sure if related but console shows this error:

```
Warning: Attempted to synchronously unmount a root while React was already rendering. React cannot finish unmounting the root until the current render has completed, which may lead to a race condition.
    at Lumino (http://localhost:3000/static/js/bundle.js:7057:60)
    at div
    at O (http://localhost:3000/static/js/bundle.js:264467:8)
    at div
    at O (http://localhost:3000/static/js/bundle.js:264467:8)
    at div
    at Notebook (http://localhost:3000/static/js/bundle.js:7424:74)
    at div
    at div
    at div
    at div
    at NotebookDemo (http://localhost:3000/static/js/bundle.js:5116:75)
    at Provider (http://localhost:3000/static/js/bundle.js:241520:3)
    at JupyterContextProvider (http://localhost:3000/static/js/bundle.js:9899:5)
    at div
    at O (http://localhost:3000/static/js/bundle.js:264467:8)
    at div
    at O (http://localhost:3000/static/js/bundle.js:264467:8)
    at BaseStyles (http://localhost:3000/static/js/bundle.js:161810:5)
    at Fe (http://localhost:3000/static/js/bundle.js:264387:60)
    at ThemeProvider (http://localhost:3000/static/js/bundle.js:163120:3)
    at ErrorBoundary (http://localhost:3000/static/js/bundle.js:241132:37)
    at Jupyter (http://localhost:3000/static/js/bundle.js:9545:5)
```

Note you can see this bug if you run the sample included in the repo here:

https://github.com/datalayer/jupyter-ui/blob/main/packages/react/src/examples/NotebookPathChange.tsx

## Expected behavior

I expect shift-enter to execute the cell and highlight the next cell

## Context

- Datalayer version:10.0.1
- Operating System and version: mac os
- Browser and version: chrome Version 124.0.6367.209

**Comment by wxirxn-bifrost** [comment]
I've noticed the same issue when changing the `nbformat` prop - Shift + Enter stops working, and I get the same warning showing up in my console. It also seems to happen when I unmount and re-mount the entire notebook.

**Comment by echarles** [comment]
Fixed by https://github.com/datalayer/jupyter-ui/pull/220

Closing.

@wxirxn-bifrost Please reopend if this issue is not fixed.

**Comment by wxirxn-bifrost** [comment]
@echarles When will this fix be released? 0.10.1 does not have this change. Thanks in advance!

**Comment by echarles** [comment]
@wxirxn-bifrost Thx for the ping. I have released 0.11.0 version.
